_book/03-interactive-notation/01-css-and-svg.md

```diff
@@ -0,0 +1,154 @@
+
+```xml
+<chord xml:id="c1">
+  <note/>
+  <note/>
+</chord>
+```
+
+will have a the following structure in the SVG:
+
+```xml
+<g class="chord" id="c1">
+  <g class="note"/>
+  <g class="note"/>
+</g>
+```
+
+You will notice that not only the tree structure is preserved, but also that MEI element names are passed as `@class` attribute values in the SVG elements, as well as the `@xml:id` of the MEI element as `@id` in the SVG.
+
+Thanks to this, and because SVG can be styled with CSS, it is straightforward to modify the appearance of elements and their content.
+
+Modifying the appearance can be done with a CSS file, or programmatically.
+
+### Applying CSS to the SVG
+
+In the CSS file you need to create rules to be applied to the SVG `<g>` elements - simply `g` in CSS - together with the class selector corresponding to the MEI element name. For example, `g.tempo` to modifying MEI `<tempo>` elements.
+
+```
+g.tempo {
+  // ... some CSS properties
+}
+```
+
+To modify the color, you need to change the `fill` property, and - in some cases - also the `color` property. The CSS rule will look like:
+
+```
+{
+  fill: crimson;
+  color: crimson;
+}
+```
+
+You can also select elements based on their hierarchy. For example, you can select `<artic>` within `<chord>` with:
+
+```
+g.chord g.artic {
+}
+```
+
+CSS can be animated, for example by making the colors pulsing. You need to specify an animation name with a duration and an iteration count together with corresponding key frames:
+
+```
+{
+  animation-name: pulse;
+  animation-duration: 1.0s;
+  animation-iteration-count: infinite;
+}
+
+@keyframes pulse {
+  0%   { fill: orange; }
+  50%  { fill: brown; }
+  100% { fill: orange; }
+}
+```
+
+CSS can also be used to change the opacity of an element. The opacity value can range from 0.0 (transparent) to 1.0 (normal default value).
+
+### Adjusting the style programmatically
+
+In applications, it is often useful to modify the CSS programmatically, for example in response to some user interactions.
+
+To do so, elements can be accessed by element and class name in the same way as with CSS. For example, for retrieving all rests, you can do:
+
+```js
+let rests = document.querySelectorAll('g.rest');
+```
+
+You can then loop through the list of rests with:
+
+```js
+for (let rest of rests) {
+    // you have now access to the rest one by one and can modify their style
+}
+```
+
+To modify the style of an element, you can assign the desired value to the corresponding key. For example, in order to change the color (fill) or a rest (element), you need to do:
+
+```js
+rest.style.fill = "dodgerblue";
+```
+
+#### Using custom data-* attributes
+
+The attributes in the SVG `<g>` elements corresponding to the MEI elements is not limited to the `@class` carrying the MEI element name and the MEI `@xml:id` passed as `@id`. Verovio also passed MEI `@type` values as additional `@class` in the SVG.
```

Comment:
```suggestion
The attributes in the SVG `<g>` elements corresponding to the MEI elements is not limited to the `@class` carrying the MEI element name and the MEI `@xml:id` passed as `@id`. Verovio also produces MEI `@type` values as additional `@class` in the SVG.
```

Reply:
I meant "passes". The `@type` is not produced. "converts" ?